styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java

```diff
@@ -169,6 +193,20 @@ WorkflowInstanceExecutionData executionData(WorkflowInstance workflowInstance) t
     return WorkflowInstanceExecutionData.fromEvents(events);
   }
 
+  private List<WorkflowInstanceExecutionData> executionData(
+      Set<WorkflowInstance> workflowInstancesSet) {
+    return workflowInstancesSet.parallelStream()
+        .map(workflowInstance -> {
+          try {
+            return executionData(workflowInstance);
+          } catch (IOException e) {
+            throw Throwables.propagate(e);
+          }
+        })
+        .sorted(WorkflowInstanceExecutionData.COMPARATOR)
```

Comment:
It's a hash set. But don't understand why a set? Do we expect duplicates? Using a list would do the job, right.